src/supervisor/watchers/handlers/index.ts

```diff
@@ -211,18 +212,22 @@ export async function setupInformer(
 
   informer.on(ERROR, (err) => {
     // Types from client library insists that callback is of type KubernetesObject
-    if ((err as any).code === ECONNRESET_ERROR_CODE) {
+    const code = (err as any).code || '';
+    if (RETRYABLE_NETWORK_ERRORS.includes(code)) {
       logger.debug(
-        {},
-        `informer ${ECONNRESET_ERROR_CODE} occurred, restarting informer`,
+        logContext,
+        `informer ${code} occurred, restarting informer`,
       );
 
       // Restart informer after 1sec
       setTimeout(async () => {
         await informer.start();
       }, 1000);
```

Comment:
So this returns a Promise. It doesn't throw err? If this fails, then do we retry?

Reply:
The `.start` function doesn't itself throw. Instead we'll get a event in this same error handler, so it will enter this piece of code again